baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/LambdaMethodReferenceTest.java

```diff
@@ -67,7 +68,83 @@ void testFunction() {
     }
 
     @Test
-    public void testPositive_block() {
+    void testLocalInstanceMethod() {
+        compilationHelper
+                .addSourceLines(
+                        "Test.java",
+                        "import " + Optional.class.getName() + ';',
+                        "class Test {",
+                        "  public Optional<Integer> foo(Optional<String> optional) {",
+                        "    // BUG: Diagnostic contains: Lambda should be a method reference",
+                        "    return optional.map(v -> bar(v));",
+                        "  }",
+                        "  private Integer bar(String value) {",
+                        "    return value.length();",
+                        "  }",
+                        "}")
+                .doTest();
```

Comment:
Not sure these negative tests are necessary when we have refactoring tests as well.